src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -136,9 +206,13 @@ private WindowScanOutput Scan_Integration_Core(string testAppPath, int expectedE
 
             var scanner = ScannerFactory.CreateScanner(config);
 
-            var output = ScanWithProvisionForBuildAgents(scanner);
+            var output = ScanAsyncWithProvisionForBuildAgents(scanner);
+
+            return Validate_Output(output, expectedErrorCount, expectedWindowCount);
+        }
 
-            // Validate for consistency
+        private WindowScanOutput Validate_Output(IReadOnlyCollection<WindowScanOutput> output, int expectedErrorCount, int expectedWindowCount = 1)
```

Comment:
We generally reserve Snake_Case for the test method names and use PascalCase for everything else